common/raw_hashtable_metadata_group.h

```diff
@@ -329,17 +337,51 @@ class MetadataGroup : public Printable<MetadataGroup> {
   // The main API for this class. This API will switch between a portable and
   // SIMD implementation based on what is most efficient, but in debug builds
   // will cross check that the implementations do not diverge.
+
+  // Load and return a group of metadata bytes out of the main metadata array at
+  // a particular `index`. The index must be a multiple of `GroupSize`. This
+  // will arrange for the load to place the group into the correct structure for
+  // efficient register-based processing.
   static auto Load(const uint8_t* metadata, ssize_t index) -> MetadataGroup;
+
+  // Store this metadata group into the main metadata array at the provided
+  // `index`. The index must be a multiple of `GroupSize`.
   auto Store(uint8_t* metadata, ssize_t index) const -> void;
```

Comment:
They are both the top-level metadata array here. I think this was at one point confused and different at a lower level (`SIMDStore` I think), but I think I've fixed it so that now the `metadata` pointer is always to the entire array, and the `index` is the index into this array of the start of the group, indexing in *bytes* rather than *groups*. PTAL.